### PR TITLE
check_package_version added, base variables for checks

### PR DESCRIPTION
**Pull request by davidbloss**

### CODE REVIEW

File: modules/check/variables.tf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+variable "check_type" {
+  type        = string
+  description = "The type of the check the module should create."
+  validation {
+    condition     = contains(["manual", null], var.check_type)
+    error_message = format("Invalid check type.  Valid values are: %s", join(",", var.check_types))
+  }
+  default = null
+}
+
```
Comment (rocktavious) [inline modules/check/variables.tf:12]:
```suggestion
locals {
  check_types = ["manual", "package_version"]
}

variable "type" {
  type        = string
  description = "The type check to create."
  validation {
    condition = contains(check_types, var.type)
    error_message = format("expected type to be one of %v", check_types)
  }
}

```

null is not a valid option - we need this to ALWAYS be specified

File: modules/check/check_package_version.tf [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  type = object({
+    type  = string
+    value = optional(string)
+  })
+  description = "A condition that should be satisfied."
+}
+
+resource "opslevel_check_package_version" "this" {
+  name     = var.name
+  enabled  = var.enabled
```
Comment (rocktavious) [inline modules/check/check_package_version.tf:37]:
```suggestion
resource "opslevel_check_package_version" "this" {
  count = var.check_type == "package_version" ? 1 : 0

  name     = var.name
  enabled  = var.enabled
```

File: versions.tf [hunk 3]
```diff
@@ -0,0 +1 @@
+/Users/davidbloss/opslevel/terraform-provider-opslevel/main/workspace/versions.tf
```
Comment (rocktavious) [inline versions.tf:1]: this path seems wrong

Comment (davidbloss) [reply versions.tf:1]: it is wrong.... 😬

File: modules/check/variables.tf [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+locals {
+  check_types = ["manual", "package_version"]
+}
+
+variable "check_type" {
+  type        = string
+  description = "The type of check to create."
+  validation {
+    condition     = contains(local.check_types, var.type)
```
Comment (rocktavious) [inline modules/check/variables.tf:11]:
```suggestion
    condition     = contains(local.check_types, var.check_type)
```

My bad

**Review by rocktavious — APPROVED** [review]
Overall its fine but we are losing some UX.

File: modules/rubric_level/main.tf [hunk 5]
```diff
@@ -0,0 +1,8 @@
+data "opslevel_rubric_levels" "all" {}
+
+data "opslevel_rubric_level" "greatest_level_by_index" {
```
[the remark this answers is not in the document]
Comment (rocktavious) [reply modules/rubric_level/main.tf:3]: I see it now over in https://github.com/OpsLevel/terraform-provider-opslevel/pull/481

Does the test HCL files allow for us to put stuff like this in there?  or is this used by more then just 1 test?

[the remark this answers is not in the document]
Comment (davidbloss) [reply modules/rubric_level/main.tf:3]: It's used for testing downstream. But it should be removed here and pushed over into the test logic there

File: modules/check/base_variables.tf [hunk 6]
```diff
@@ -0,0 +1,44 @@
+variable "category" {
+  type        = string
+  description = "The id of the category the check belongs to."
```
[the remark this answers is not in the document]
Comment (davidbloss) [reply modules/check/base_variables.tf:3]: ohhh yes I see now. I'll bring back `rubric.tf` and make it work

**Pull request merged**